incubator/hnc/pkg/controllers/object_controller.go

```diff
@@ -284,14 +291,31 @@ func (r *ObjectReconciler) onDelete(ctx context.Context, log logr.Logger, nnm ty
 	return nil
 }
 
+func (r *ObjectReconciler) ensureCanPropagate(nsNm string, inst *unstructured.Unstructured) bool {
+	r.Forest.Lock()
+	defer r.Forest.Unlock()
+	ns := r.Forest.Get(nsNm)
+	if !ns.Exists() {
+		return false
+	}
+	if inst.GetDeletionGracePeriodSeconds() != nil {
+		ns.SetCondition(fmt.Sprint(r.GVK, ns, nsNm), tenancy.ObjectInternal, "finalizer set")
+		return false
+	}
+	if inst.GetDeletionTimestamp() != nil {
+		ns.SetCondition(fmt.Sprint(r.GVK, ns, nsNm), tenancy.ObjectInternal, "finalizer set")
```

Comment:
Ditto. Also, rather than cutting-and-pasting the complex `fmt.Sprint` line, can you factor that out somehow?

Reply:
Factor that out meaning declare it as a local variable and use that as opposed to calling it all on one line and making it hard to read? Sure!

Reply:
Yup, that works!